### PR TITLE
Nested Auth - Interactive & silent request [msal]

### PR DESCRIPTION
**Pull request by juan-arias**
## Proposed changes

Changes to support MetaOS when doing an interactive & silent request.

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

**Review by Veena11 — APPROVED** [review]

File: MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h [hunk 1]
```diff
@@ -110,7 +116,22 @@ to target MSAL at a specific test slice & flight. These apply to all requests ma
  */
 - (nonnull instancetype)initWithClientId:(NSString *)clientId
                              redirectUri:(nullable NSString *)redirectUri
-                               authority:(nullable MSALAuthority *)authority NS_DESIGNATED_INITIALIZER;
+                               authority:(nullable MSALAuthority *)authority DEPRECATED_MSG_ATTRIBUTE("Use initWithClientId:redirectUri:authority:nestedAuthBrokerClientId:nestedAuthBrokerRedirectUri instead");
```
Comment (oldalton) [inline MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h:119]: We shouldn't be deprecating this and forcing everyone to pass nested uris

File: MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h [hunk 2]
```diff
@@ -110,7 +116,22 @@ to target MSAL at a specific test slice & flight. These apply to all requests ma
...
+ @param  clientId       The clientID of your application, you should get this from the app portal.
+ @param  redirectUri    The redirect URI of the application
+ @param  authority      The target authority
+ @param  nestedClientId     The clientID of your child application
+ @param  nestedRedirectUri    The redirect URI of the child application
+ */
+- (nonnull instancetype)initWithClientId:(NSString *)clientId
+                             redirectUri:(nullable NSString *)redirectUri
+                               authority:(nullable MSALAuthority *)authority
+                          nestedClientId:(nullable NSString *)nestedClientId
```
Comment (oldalton) [inline MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h:133]: Let's call those consistently, should be "nestedAuthBrokerClientId"

**Review by oldalton — CHANGES_REQUESTED** [review]

File: MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h (hunk 1 above)
Comment (juan-arias) [reply MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h]: Makes sense, updated.

File: MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h (hunk 2 above)
Comment (juan-arias) [reply MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h]: I missed them, updated.

File: MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h [hunk 3]
```diff
@@ -110,7 +116,22 @@ to target MSAL at a specific test slice & flight. These apply to all requests ma
...
+ @param  redirectUri    The redirect URI of the application
+ @param  authority      The target authority
+ @param  nestedAuthBrokerClientId     The clientID of your child application
+ @param  nestedAuthBrokerRedirectUri    The redirect URI of the child application
+ */
+- (nonnull instancetype)initWithClientId:(NSString *)clientId
+                             redirectUri:(nullable NSString *)redirectUri
+                               authority:(nullable MSALAuthority *)authority
+                nestedAuthBrokerClientId:(nullable NSString *)nestedAuthBrokerClientId
+             nestedAuthBrokerRedirectUri:(nullable NSString *)nestedAuthBrokerRedirectUri NS_DESIGNATED_INITIALIZER;
```
Comment (oldalton) [inline MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h:134]: I don't think this should be a designated initializer. I don't expect any 3P app to use this API

**Review by oldalton — CHANGES_REQUESTED** [review]

File: MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h (hunk 3 above)
Comment (juan-arias) [reply MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h]: Updated

**Pull request merged**